### PR TITLE
example program for paper intro

### PR DESCRIPTION
**Pull request by migeed-z**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83945

### CODE REVIEW

File: test/fx/test_z3_gradual_types.py [hunk 1]
```diff
@@ -1463,6 +1463,45 @@ def forward(self, x: TensorType([20, 20])):
...
 class TestSingleOperation(unittest.TestCase):
+
+    def test_conv_wrong_example(self):
+        class BasicBlock(torch.nn.Module):
+            def __init__(self):
+                super(BasicBlock, self).__init__()
+                self.conv1 = torch.nn.Conv2d(in_channels=2, out_channels=2,
+                                             kernel_size=2, stride=2,
+                                             padding=2, groups=2, bias=False, dilation=2)
+
```
Comment (jansel) [inline test/fx/test_z3_gradual_types.py:1474]: It might make sense to add a torch.  A torch.nn.ReLU() after each Conv2 to make this more realistic.

File: test/fx/test_z3_gradual_types.py [hunk 2]
```diff
@@ -1463,6 +1463,45 @@ def forward(self, x: TensorType([20, 20])):
...
+                self.conv2 = torch.nn.Conv2d(in_channels=4, out_channels=2,
+                                             kernel_size=2, stride=2,
+                                             padding=2, groups=2, bias=False, dilation=2)
+
+            def forward(self, x: Dyn):
```
Comment (jansel) [inline test/fx/test_z3_gradual_types.py:1479]: We should shorten this to 
```
example_nn = torch.nn.Sequential(
    torch.nn.Conv2d(in_channels=2, out_channels=2,
                    kernel_size=2, stride=2,
                    padding=2, groups=2, bias=False, dilation=2),
    torch.nn.ReLU(),
    torch.nn.Conv2d(in_channels=4, out_channels=2,
                    kernel_size=2, stride=2,
                    padding=2, groups=2, bias=False, dilation=2)
    torch.nn.ReLU(),
)
```

File: test/fx/test_z3_gradual_types.py (hunk 1 above)
Comment (migeed-z) [reply test/fx/test_z3_gradual_types.py:1474]: Added a relu after every conv

File: test/fx/test_z3_gradual_types.py (hunk 2 above)
Comment (migeed-z) [reply test/fx/test_z3_gradual_types.py:1481]: But I think this is composing all the functions together. Can we keep it as the original format so we can have two separate calls for x? this allows us to have a satisfiable assignment, but one that could only be Dyn.

**Review by jansel — APPROVED** [review]

**Comment by migeed-z** [comment]
@pytorch merge -g
